pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -37,6 +37,10 @@ import (
 // maxUpdateRetries is the maximum number of retries used for update conflict resolution prior to failure
 const maxUpdateRetries = 10
 
+// StatefulSetPodNameLabel is label added to each Pod in a StatefulSet. Its purpose is to allow Services to select
+// individual Pods in the StatefulSet.
+const StatefulSetPodNameLabel = "statefulSetPodName"
```

Comment:
Do we have conventions for machine-managed label names? For controller revision, we use `controller.kubernetes.io/hash`. For failure domains, we use `failure-domain.beta.kubernetes.io/region`. For Deployment, we use `pod-template-hash`.

Also, should we put this constant in `types.go` for a given API version like the others?

Reply:
> Also, should we put this constant in types.go for a given API version like the others?
The label is not version specific, and, by convention, the versioning seems to only apply to annotations. If we were to move it, I think the only other appropriate home would well_know_labels.go in the apis package.

I just want the name to be easily usable `statefulset.kubernetes.io/podName` is fine with me, but as you point out we don't have conventions. It would be good to start a trend though. Do you have thoughts on a better name?